packages/ui5-application-writer/src/data/defaults.ts

```diff
@@ -57,36 +58,41 @@ export const UI5_DEFAULT = {
  * @param {UI5} [ui5] - the UI5 instance
  * @returns {UI5} the updated UI5 instance
  */
-export function mergeUi5(ui5?: UI5): UI5 {
+export function mergeUi5(ui5: Partial<UI5>): UI5 {
     const merged: Partial<UI5> = {
-        minUI5Version: ui5?.minUI5Version || '1.60',
-        version: ui5?.version || UI5_DEFAULT.DEFAULT_UI5_VERSION // no version indicates the latest available should be used
+        minUI5Version: ui5.minUI5Version || UI5_DEFAULT.MIN_UI5_VERSION,
+        version: ui5.version || UI5_DEFAULT.DEFAULT_UI5_VERSION // no version indicates the latest available should be used
     };
-    merged.framework = ui5?.framework || 'SAPUI5';
+    merged.framework = ui5.framework || 'SAPUI5';
     merged.frameworkUrl =
-        ui5?.frameworkUrl || merged.framework === 'SAPUI5' ? UI5_DEFAULT.SAPUI5_CDN : UI5_DEFAULT.OPENUI5_CDN;
+        ui5.frameworkUrl || merged.framework === 'SAPUI5' ? UI5_DEFAULT.SAPUI5_CDN : UI5_DEFAULT.OPENUI5_CDN;
 
     // if a specific local version is provided, use it, otherwise, sync with version but keep minimum versions in mind
-    if (ui5?.localVersion) {
+    if (ui5.localVersion) {
         merged.localVersion = ui5.localVersion;
     } else {
-        merged.localVersion =
-            // eslint-disable-next-line no-nested-ternary
-            merged.version === UI5_DEFAULT.DEFAULT_UI5_VERSION
-                ? UI5_DEFAULT.DEFAULT_LOCAL_UI5_VERSION
-                : merged.framework === 'SAPUI5'
+        if (merged.version === UI5_DEFAULT.DEFAULT_UI5_VERSION) {
+            merged.localVersion = UI5_DEFAULT.DEFAULT_LOCAL_UI5_VERSION;
+        } else {
+            merged.localVersion =
+                merged.framework === 'SAPUI5'
                     ? UI5_DEFAULT.MIN_LOCAL_SAPUI5_VERSION
                     : UI5_DEFAULT.MIN_LOCAL_OPENUI5_VERSION; // minimum version available as local libs
+        }
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
I've refactored this function to remove the 3 comments disabling eslint rules, removed the non-null assertions and brought down the cognitive complexity as reported by Sonar (original -> 17) - https://github.com/SAP/open-ux-tools/pull/144/commits/58cf2e899a3acc9d084d06f26621afa401c96cc6

@tobiasqueck, let me know if you're okay with this change. We can revert it if it's better to do this in a different PR. Cheers.